src/core/expr/fexpr_cumsum.cc

```diff
@@ -23,6 +23,30 @@ class Column_cumsum : public Virtual_ColumnImpl {
       xassert(col_.can_be_read_as<T>());
     }
 
+    void _materialize(Column& out, bool) const {
+      Column newcol = Column::new_data_column(nrows_, stype());
+      T* data = static_cast<T*>(newcol.get_data_editable(0));
+      parallel_for_static(nrows_,
+          [&](size_t i) {
+            data[i] = (i == 0) ? data[i] : data[i] + data[i - 1];
+          });
+        out = newcol;
+      }
```

Comment:
This will not work, because you never use the data from `col_` and this is the column you should calculate running total for.